tests/src/prompt.cpp

```diff
@@ -0,0 +1,32 @@
+#include <catch2/catch.hpp>
+
+/*
+ * Catch2: examples
+ * for more examples checkout:
+ *   https://github.com/catchorg/Catch2/tree/devel/examples
+ *
+ * Catch has two natural expression assertion macro's:
+ * - REQUIRE() stops at first failure.
+ * - CHECK() continues after failure.
+ *
+ * There are two variants to support decomposing negated expressions:
+ * - REQUIRE_FALSE() stops at first failure.
+ * - CHECK_FALSE() continues after failure.
+ *
+ * TEST_CASE( "Title here", "[short name]" ) {
+ *     REQUIRE( [BOOLEAN CONDITION] ) ;
+ * }
+ */
+
+extern "C" {
+# include "prompt.h"
+}
+
+TEST_CASE( "create_destroy prompt", "[prompt]" ) {
+	t_prompt *prompt;
+	prompt = prompt_create("username", "hostname");
+	REQUIRE(prompt != NULL);
+	prompt_print(prompt);
+	prompt_destroy(prompt);
```

Comment:
looks good man, maybe you could have prompt_destroy take a t_prompt ** so you can free the memory and set the prompt * to NULL (and return NULL). so you can use the prompt_destroy() function to return NULL and save some lines